src/DynamoCoreWpf/Views/Core/AnnotationView.xaml.cs

```diff
@@ -276,5 +274,63 @@ private void OnGraphLayoutAnnotation(object sender, RoutedEventArgs e)
             }
         }
 
+        private void GroupDescriptionTextBox_OnIsVisibleChanged(object sender, DependencyPropertyChangedEventArgs e)
+        {
+            var textbox = sender as TextBox;
+            if (textbox == null || textbox.Visibility != Visibility.Visible) return;
+
+            //Record the value here, this is useful when title is popped from stack during undo
+            ViewModel.WorkspaceViewModel.DynamoViewModel.ExecuteCommand(
+                new DynCmd.UpdateModelValueCommand(
+                    Guid.Empty, ViewModel.AnnotationModel.GUID, "GroupNameTextBlockText",
+                    GroupDescriptionTextBox.Text));
+
+            ViewModel.WorkspaceViewModel.DynamoViewModel.RaiseCanExecuteUndoRedo();
+
+            textbox.Focus();
+            if (textbox.Text.Equals(Properties.Resources.GroupDefaultText))
+            {
+                textbox.SelectAll();
+            }
+        }
+
+        private void GroupDescriptionTextBox_GotFocus(object sender, RoutedEventArgs e)
+        {
+            this.GroupDescriptionTextBox.CaretIndex = Int32.MaxValue;
+        }
+
+        private void GroupDescriptionTextBox_TextChanged(object sender, TextChangedEventArgs e)
+        {
+            if (ViewModel != null)
```

Comment:
Been told by Radu that it is preferred to do `if(ViewModel is null) return;` rather than this.

Reply:
yep good catch, thanks!